addons/sourcemod/scripting/shop/functions.sp

```diff
@@ -113,9 +113,12 @@ public void Functions_OnConVarChange(ConVar convar, const char[] oldValue, const
 void Functions_UnregisterMe(Handle plugin)
 {
 	int index = -1;
-	while ((index = FindValueInArray(g_hFuncArray, plugin)) != -1)
+	while ((index = g_hFuncArray.FindValue(plugin)) != -1)
 	{
-		RemoveFromArray(g_hFuncArray, index);
+		delete view_as<Handle>(g_hFuncArray.Get(index+1));
+
+		g_hFuncArray.Erase(index);
+		g_hFuncArray.Erase(index);
 	}
 }
```

Comment:
I think all this function can be removed, and used public function Functions_RemoveFromMenuNative.
We can pass parameters in this function only plugin handle. It looks like this:
```hpp
public int Native_UnregisterMe(Handle plugin, int params)
{
 // ....
  Functions_RemoveFromMenuNative(plugin, params); // but we can keep 0 parameters instead of `params`, they not relevant
}
```